webcompat/form.py

```diff
@@ -111,12 +206,72 @@ class IssueForm(FlaskForm):
     submit_type = HiddenField()
 
 
-def get_form(form_data):
+class PrefixedRadioField(RadioField):
+    """Prefix radio field label with an image."""
+    def __init__(self, *args, **kwargs):
+        prefixed_choices = kwargs.pop('choices')
+        template = '<div class={cls}><img src={src}/></div> {text}'
+        choices = []
+
+        for slug, img, text in prefixed_choices:
+            cls = 'icon-container'
+            filename = 'img/svg/icons/{img}'.format(img=img)
+            src = url_for('static', filename=filename)
+            label = Markup(template.format(src=src, cls=cls, text=text))
+            choice = (slug, label)
+            choices.append(choice)
+
+        kwargs['choices'] = choices
+        super(PrefixedRadioField, self).__init__(*args, **kwargs)
+
+
+class IssueFormRedesigned(IssueForm):
+    """Re-designed version of IssueForm:
+    Define form fields and validation for our bug reporting form.
+    """
+
+    steps = NEW_ISSUE_STEPS
+
+    browser = StringField(u'Browser', [Optional()])
+    device = StringField(u'Device', [Optional()])
+    os = StringField(u'Operating System', [Optional()])
+
+    problem_category = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=problem_choices_redesign
+    )
+    other_problem = StringField(
+        other_problem_label,
+        [InputRequired(message=other_problem_message)]
+    )
+    other_browser = StringField(
+        other_browser_label,
+        [InputRequired(message=other_browser_message)]
+    )
+    site_bug_subcategory = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=site_bug_choices
+    )
+    layout_bug_subcategory = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=layout_bug_choices
+    )
+    video_bug_subcategory = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=video_bug_choices
+    )
+    browsers = PrefixedRadioField(
+        [InputRequired(message=radio_message)],
+        choices=browser_choices
+    )
+
+
+def get_form(form_data, form=IssueForm):
```

Comment:
nit: In the future, this is probably a welcome improvement. #tofixlater
There could be a separate PR which basically give structure to the code for things without AB testing in a way that make it easier to configure. 

Thanks!

Reply:
Noted for the issues to be filled for future reference.